runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -5381,6 +5381,257 @@ bool genInstanceOfOrCheckCastNullTest(TR::Node* node, TR::CodeGenerator* cg, TR:
       return isNullTestImplicit;
    }
 
+
+/** \brief
+ *     Generates a helper call for \p node.
+ *
+ *  \details
+ *     Function switches to outlined section and generate a JIT helper dispatch sequence.
+ *     It is responsibility of the caller to switch back to main line.
```

Comment:
This seems like an odd design choice? If the caller is responsible for switching back to the mainline would it be more intuitive for the caller to be responsible to switch to the OOL path to begin with?

Reply:
While writing this I wanted to use this as a generic function to generate helper call. So all `DUAL_MODE_HELPER` and `fast_path_jitInstanceOf` who uses J9S390CHelperLinkage can take advantage of this function so tried to cover following cases. 
1. Helper call is within ICF in OOL that means we have already switched to OOL and this function just need to generate helper call. 
2. There is only helper call in OOL means so this function can switch to OOL. 
In all cases, the way we return to main line is different and that is why I designed it this way. 
I am OK in refactoring this and keeping switching from MainLine to  OOL and vice versa in caller.

Reply:
I only see a single use case of this function. To me it seems counter intuitive that a function performs something partially and relies on the caller to finish off the logic, especially when this partial logic can be moved to the caller anyway. There is no coupling between generating an OOL code section and creating a helper call. I would prefer for the OOL logic to be factored out to the caller which swaps back from the OOL and avoid this partial logic.